openslides_backend/services/database/database_writer.py

```diff
@@ -547,28 +548,29 @@ def execute_sql(
             )
         except CheckViolation as e:
             _, table, _, constraint_name, _ = e.args[0].split('"')
-            record = False
+            # Fetch the generated constraint from the initially applied schema.
+            in_table_block = False
             with open("meta/dev/sql/schema_relational.sql") as f:
                 for line in f:
                     # search only in the table block to prevent finding duplicates of other tables
-                    if "CREATE TABLE" in line and table in line:
-                        record = True
-                    elif record and constraint_name in line:
+                    if f"CREATE TABLE {table}" in line:
+                        in_table_block = True
+                    elif in_table_block and constraint_name in line:
```

Comment:
1. Table name could still be substring of another table string if the end is not also checked against an empty space
2. There is no guarantee that the table create lines are always going to have the format `CREATE TABLE <table_name>`. In the future it might be necessary to use something like `CREATE TABLE IF NOT EXISTS`. Maybe use regex?
3. The `constraint_name` line has the same problem as the table line